### PR TITLE
feat(GaussDB): add gaussdb opengauss database resource

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  add gaussdb opengauss database resource
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  add gaussdb opengauss database resource
```

## PR Checklist

* [ ] Tests added/passed.

```
make testacc TEST=./huaweicloud/services/acceptance/gaussdb/ TESTARGS='-run TestAccOpenGaussDatabase_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb/ -v -run TestAccOpenGaussDatabase_basic -timeout 360m -parallel 4
=== RUN   TestAccOpenGaussDatabase_basic
=== PAUSE TestAccOpenGaussDatabase_basic
=== CONT  TestAccOpenGaussDatabase_basic
--- PASS: TestAccOpenGaussDatabase_basic (1954.20s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   1954.246s
```

* [ ] Documentation updated.
* [ ] Schema updated.
* [x] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
删除了
    ab. Related resources (parent resources) not found
实例删除了
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
删除后再删除
    bb. Related resources (parent resources) not found
    实例删了再删除

### CODE REVIEW

File: huaweicloud/services/acceptance/gaussdb/resource_huaweicloud_gaussdb_opengauss_database_test.go [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+}
+
+func testAccOpenGaussDatabaseImportStateFunc(name string) resource.ImportStateIdFunc {
+	return func(s *terraform.State) (string, error) {
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return "", fmt.Errorf("resource (%s) not found: %s", name, rs)
+		}
+		if rs.Primary.Attributes["instance_id"] == "" || rs.Primary.Attributes["name"] == "" {
+			return "", fmt.Errorf("resource (%s) not found: %s", name, rs)
```
Comment (saf3dfsa) [inline huaweicloud/services/acceptance/gaussdb/resource_huaweicloud_gaussdb_opengauss_database_test.go]: The error info is wrong.

File: huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_opengauss_database.go [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `Specifies the database classification.`,
+			},
+			"size": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Specifies the database classification.`,
```
Comment (saf3dfsa) [inline huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_opengauss_database.go]:
```suggestion
				Description: `Indicates the database size.`,
```

File: huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_opengauss_database.go [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+			},
+			"compatibility_type": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Specifies the database classification.`,
```
Comment (saf3dfsa) [inline huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_opengauss_database.go]:
```suggestion
				Description: `Indicates the database compatibility type.`,
```

File: huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_opengauss_database.go [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+		}
+	}
+	return err
+}
+
+func resourceOpenGaussDatabaseImportState(_ context.Context, d *schema.ResourceData, _ interface{}) ([]*schema.ResourceData,
+	error) {
+	parts := strings.Split(d.Id(), "/")
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("invalid format specified for import id, must be <instance_id>/<name>")
```
Comment (saf3dfsa) [inline huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_opengauss_database.go]:
```suggestion
		return nil, fmt.Errorf("invalid format specified for import ID, must be <instance_id>/<name>")
```

File: docs/resources/gaussdb_opengauss_database.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+  Changing this parameter will create a new resource.
+
+* `owner` - (Optional, String, ForceNew) Specifies the Database user. Defaults to **root**. The value must be an existing
+  username and must be different from system usernames. System users: **rdsAdmin**, **rdsMetric**, **rdsBackup** and
+  **rdsRepl**.
+
+  Changing this parameter will create a new resource.
+
+* `character_set` - (Optional, String, ForceNew) Specifies the database character set. Defaults to **UTF8**.
```
Comment (Jason-Zhang9309) [inline docs/resources/gaussdb_opengauss_database.md]: same to line 42, duplicate parameters?

File: huaweicloud/services/acceptance/gaussdb/resource_huaweicloud_gaussdb_opengauss_database_test.go [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+		offset++
+	}
+	if database == nil {
+		return nil, golangsdk.ErrDefault404{}
+	}
+	return database, nil
+}
+
+func buildOpenGaussDatabaseQueryParams(offset int) string {
+	return fmt.Sprintf("?limit=100&offset=%v", offset)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/acceptance/gaussdb/resource_huaweicloud_gaussdb_opengauss_database_test.go:70]: should like this?
```suggestion
	return fmt.Sprintf("?limit=100&offset=%v", offset * 100)
```

File: huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_opengauss_database.go [hunk 7]
```diff
@@ -0,0 +1,314 @@
...
+		d.Set("lc_ctype", utils.PathSearch("datctype", database, nil)),
+		d.Set("size", utils.PathSearch("size", database, nil)),
+		d.Set("compatibility_type", utils.PathSearch("compatibility_type", database, nil)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func buildOpenGaussDatabaseQueryParams(offset int) string {
+	return fmt.Sprintf("?limit=100&offset=%v", offset)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_opengauss_database.go:236]: should it be like this?
```suggestion
	return fmt.Sprintf("?limit=100&offset=%v", offset * 100)
```

File: docs/resources/gaussdb_opengauss_database.md (hunk 5 above)
Comment (houpeng80) [reply docs/resources/gaussdb_opengauss_database.md]: done

**Pull request merged**